src/Raven.Client/Documents/Commands/WaitForDatabaseIndexNotificationCommand.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Raven.Client.Http;
+using Sparrow.Json;
+
+namespace Raven.Client.Documents.Commands
+{
+    public class WaitForDatabaseIndexNotificationCommand : RavenCommand
+    {
+        private readonly long _index;
+        private readonly string _database;
+
+        public WaitForDatabaseIndexNotificationCommand(long index, string database)
+        {
+            _index = index;
+            _database = database;
+        }
+
+        public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
+        {
+            url = $"{node.Url}/databases/{_database}/admin/rachis/wait-for-index-notification?index={_index}";
+            var request = new HttpRequestMessage
+            {
+                Method = HttpMethod.Post
+            };
+            return request;
+        }
+
+        public override void SetResponse(JsonOperationContext context, BlittableJsonReaderObject response, bool fromCache)
+        {
+            if (response == null)
+                ThrowInvalidResponse();
+        }
+
+        public override bool IsReadRequest => true;
```

Comment:
@ppekrol IIRC the `IsReadRequest` is misleading here and basically means if we can read it from cache., right?
For this command I would say we never want to read it from cache, so it should return `false`

Reply:
Yes, cache and broadcast related IIRC